pkg/broker/awsbroker.go

```diff
@@ -357,9 +366,21 @@ func (db Db) ServiceDefinitionToOsb(sd CfnTemplate) osb.Service {
 }
 
 func (b *AwsBroker) generateS3HTTPUrl(serviceDefName string) *string {
-	prefix := "https://s3.amazonaws.com/"
-	if b.s3region != "us-east-1" {
-		prefix = fmt.Sprintf("https://s3-%s.amazonaws.com/", b.s3region)
+	if b.partition == "aws" {
+		// AWS Standard Partition
+		prefix := "https://s3.amazonaws.com/"
+		if b.s3region != "us-east-1" {
+			prefix = fmt.Sprintf("https://s3-%s.amazonaws.com/", b.s3region)
+		}
+		return aws.String(prefix + b.s3bucket + "/" + b.s3key + strings.TrimSuffix(serviceDefName, "-apb") + b.templatefilter)
+	} else if b.partition == "aws-cn" {
+		// AWS China Partition
+		objKey := b.s3key + strings.TrimSuffix(serviceDefName, "-apb") + b.templatefilter
+		objURL := fmt.Sprintf("https://%s.s3.%s.amazonaws.com.cn/%s", b.s3bucket, b.s3region, objKey)
+		return aws.String(objURL)
+	} else {
+		// AWS GovCloud Partition
+		// TODO add implementation for GovCloud partition
```

Comment:
URL scheme for govcloud is the same as the `aws` partition (in most cases), so to retain current govcloud support can we invert this to be more like:

```go
if b.partition == "aws-cn" {
    // AWS China Partition
    ...
} else {
    // all other partitions use the default URL scheme
    ...
}
```

Reply:
sure